runtime/src/deterministic.rs

```diff
@@ -997,18 +1002,15 @@ pub struct Sink {
     auditor: Arc<Auditor>,
     me: SocketAddr,
     peer: SocketAddr,
-    sender: mpsc::UnboundedSender<Bytes>,
+    sender: ByteSink,
 }
 
 impl crate::Sink for Sink {
-    async fn send(&mut self, msg: Bytes) -> Result<(), Error> {
-        let len = msg.len();
-        self.auditor.send(self.me, self.peer, msg.clone());
-        self.sender
-            .send(msg)
-            .await
-            .map_err(|_| Error::WriteFailed)?;
-        self.metrics.network_bandwidth.inc_by(len as u64);
+    async fn send(&mut self, msg: &[u8]) -> Result<(), Error> {
+        self.auditor.send(self.me, self.peer, Bytes::copy_from_slice(msg));
```

Comment:
nit: I'd recommend modifying the `auditor` method to take `&[u8]` to avoid this copy (underneath the hood, we are just feeding this into a hasher: https://github.com/commonwarexyz/monorepo/blob/88947aa3791e020cb6d3e3af0c7d4696a52d8c2f/runtime/src/deterministic.rs#L219C32-L219C33).